docs/source/tutorials/SAR/SAR tutorial.ipynb

```diff
@@ -0,0 +1,1180 @@
+    "    \n",
+    "    ica = FastICA()\n",
+    "    pcaica_act = ica.fit_transform(pca_act)\n",
+    "    \n",
+    "    normalizer = MinMaxScaler((-1, 1))    \n",
+    "    normalizer.fit(pcaica_act)\n",
+    "    \n",
+    "    if save:\n",
+    "        joblib.dump(ica, 'ica_demo.pkl') \n",
+    "        joblib.dump(pca, 'pca_demo.pkl')\n",
+    "        joblib.dump(normalizer, 'scaler_demo.pkl')\n",
+    "    \n",
+    "    return ica, pca, normalizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d60f8d5",
+   "metadata": {},
+   "source": [
+    "### Train with SAR on a more challenging task"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3c987cc",
+   "metadata": {},
+   "source": [
+    "In order to train with SAR, we initialize a `SynNoSynWrapper`, which implements the policy architecture presented below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d499c17",
+   "metadata": {},
+   "source": [
+    "<div style=\"width:100%;\">\n",
+    "    <img src=\"./SAR_images/pol_arch.png\" alt=\"Policy Architecture\" style=\"width:700px; display:block; margin-left:auto; margin-right:auto;\"/>\n",
+    "</div."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "997767da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class SynNoSynWrapper(gym.ActionWrapper):\n",
+    "    \"\"\"\n",
+    "    gym.ActionWrapper that reformulates the action space as the combination of a task-general synergy space and a\n",
+    "    task-specific orginal space, and uses this mix to step the environment in the original action space.\n",
+    "    \"\"\"\n",
+    "    def __init__(self, env, ica, pca, scaler, phi):\n",
+    "        super().__init__(env)\n",
+    "        self.ica = ica\n",
+    "        self.pca = pca\n",
+    "        self.scaler = scaler\n",
+    "        self.weight = phi\n",
+    "        \n",
+    "        self.syn_act_space = self.pca.components_.shape[0]\n",
+    "        self.no_syn_act_space = env.action_space.shape[0]\n",
+    "        self.full_act_space = self.syn_act_space + self.no_syn_act_space\n",
+    "        \n",
+    "        self.action_space = gym.spaces.Box(low=-1., high=1., shape=(self.full_act_space,),dtype=np.float32)\n",
+    "    def action(self, act):\n",
+    "        syn_action = act[:self.syn_act_space]\n",
+    "        no_syn_action = act[self.syn_act_space:]\n",
+    "        \n",
+    "        syn_action = self.pca.inverse_transform(self.ica.inverse_transform(self.scaler.inverse_transform([syn_action])))[0]\n",
+    "        final_action = self.weight * syn_action + (1 - self.weight) * no_syn_action\n",
+    "        \n",
+    "        return final_action"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "714d80be",
+   "metadata": {},
+   "source": [
+    "Additionally, we initialize a `SynergyWrapper`, which implements a variation of the above policy architecture that only leverages task-general synergistic activations."
```

Comment:
Everything after this line isn't rendering properly. It all slides right instead of the new line and going to the bottom

Reply:
fixed